### PR TITLE
Restrict Admin from unlinking users from other district

### LINKED ISSUE #1152: User facility unlink permission issue.
**Describe the bug**
In user's section a DistrictAdmin can unlink linked facilities of other DistricAdmins from **other states**.

**To Reproduce**

1. Signin as a DistrictAdmin
2. Go to user's section
3. Find any other user from a different state and unlink facilities from its list.
4. See result

**Expected behavior**
A district admin of a particular state should be allowed to unlink facilities of users from its own district only.

**Recording**

https://user-images.githubusercontent.com/39593226/200363539-1789bde4-fd47-4ae5-ae5e-ecee240619e3.mp4

### PR DESCRIPTION
**Pull request by Pranshu1902**
Fixes #1152 
Ensure that the admin's district is same as that of the facility user's district being deleted and also able to home facilities only for users within same district.

### CODE REVIEW

File: care/users/api/viewsets/users.py [hunk 1]
```diff
@@ -286,6 +286,9 @@ def delete_facility(self, request, *args, **kwargs):
             )
         if user.home_facility == facility:
             raise ValidationError({"facility": "Cannot Delete User's Home Facility"})
+        # ensure that district admin only able to delete in the same district
+        if user.district.name != requesting_user.district.name:
```
Comment (vigneshhari) [inline care/users/api/viewsets/users.py:290]: use id for equality checks

**Review by vigneshhari — CHANGES_REQUESTED** [review]

File: care/users/api/viewsets/users.py (hunk 1 above)
Comment (Pranshu1902) [reply care/users/api/viewsets/users.py]: Done

**Comment by siddnikh** [comment]
LGTM

**Review by siddnikh — APPROVED** [review]
LGTM

File: care/users/api/viewsets/users.py [hunk 2]
```diff
@@ -307,6 +307,9 @@ def delete_facility(self, request, *args, **kwargs):
             )
         if user.home_facility == facility:
             raise ValidationError({"facility": "Cannot Delete User's Home Facility"})
+        # ensure that district admin only able to delete in the same district
+        if user.district.id != requesting_user.district.id:
```
Comment (sainak) [inline care/users/api/viewsets/users.py:311]:
```suggestion
        if user.district_id != requesting_user.district_id:
```

**Review by sainak — CHANGES_REQUESTED** [review]

**Comment by sainak** [comment]
@Pranshu1902 add tests

**Comment by Pranshu1902** [comment]
@sainak @vigneshhari please review and merge the PR. I have added tests cases

**Review by rithviknishad — APPROVED** [review]

**Comment by sainak** [comment]
@Pranshu1902 the test are not catching the changes, try to assert the error message response

**Comment by Pranshu1902** [comment]
@sainak done

File: care/users/api/viewsets/users.py [hunk 3]
```diff
@@ -280,6 +286,7 @@ def delete_facility(self, request, *args, **kwargs):
             )
         if user.home_facility == facility:
             raise ValidationError({"facility": "Cannot Delete User's Home Facility"})
+
```
Comment (sainak) [inline care/users/api/viewsets/users.py:289]:
```suggestion
```

**Review by sainak — APPROVED** [review]

File: care/users/api/viewsets/users.py [hunk 4]
```diff
@@ -253,6 +253,12 @@ def clear_home_facility(self, request, *args, **kwargs):
         if not self.has_user_type_permission_elevation(requesting_user, user):
             raise ValidationError({"home_facility": "Cannot Access Higher Level User"})
 
+        # ensure that district admin only able to delete in the same district
+        if user.district_id != requesting_user.district_id:
```
Comment (vigneshhari) [inline care/users/api/viewsets/users.py:257]: A state admin need not have the same district as the requesting user right? Superadmin would not need any checks  as well.

**Review by vigneshhari — CHANGES_REQUESTED** [review]

File: care/users/api/viewsets/users.py (hunk 4 above)
Comment (Pranshu1902) [reply care/users/api/viewsets/users.py]: @vigneshhari done, now this will only check for users equal to or below District Admin's position

**Review by sainak — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**